src/components/SubNavigation/hooks/useVisibility.ts

```diff
@@ -0,0 +1,78 @@
+import {useCallback, useEffect, useState} from 'react';
+
+const useVisibility = (
+    menuOpened: boolean,
+    miniTocOpen: boolean,
+): [boolean, (value: boolean) => void] => {
+    const [visible, setVisibilityLocal] = useState(true);
+    const [hiddingTimeout, setHiddingTimeout] = useState<number | undefined>(undefined);
+    const [lastScrollY, setLastScrollY] = useState(
+        typeof window === 'undefined' ? null : window.scrollY,
+    );
+
+    const controlVisibility = useCallback(() => {
+        const scrollY = window.scrollY;
+
+        if (lastScrollY === null) {
+            return setLastScrollY(scrollY);
+        }
+
+        if (miniTocOpen || menuOpened || scrollY < 55) {
```

Comment:
you need to give names to the numerical variables so that it’s clear what 55 50 etc. is